ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,400 @@
+            f'{self.parameterset.mask}': '/data/model_mask.nc',
+            f'{self.forcing_dir}': '/data/forcing',
+            f'{self.work_dir}': '/output',
+        }
+
+        if CFG['container_engine'].lower() == 'singularity':
+            args = [
+                "singularity",
+                "exec",
+            ]
+            args += ["--bind", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(singularity_image)
+
+        elif CFG['container_engine'].lower() == 'docker':
+            args = [
+                "docker",
+                "run -ti",
+            ]
+            args += ["--volume", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(docker_image)
+
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+
+        lisvap_filename = Path(lisvap_file).name
+        args += ['python3', '/opt/Lisvap/src/lisvap1.py', f"/output/{lisvap_filename}"]
+        container = subprocess.Popen(args, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        exit_code = container.wait()
+        stdout, stderr = container.communicate()
+        return exit_code, stdout, stderr
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_x(grid),
+                "latitude": self.bmi.get_grid_y(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da
+
+    @property
+    def parameters(self) -> Iterable[Tuple[str, Any]]:
+        """List the parameters for this model."""
+        if self.forcing_dir and self.work_dir:
+            return {
+                'forcing_dir': self.forcing_dir,
+                'work_dir': self.work_dir,
+            }.items()
+        return []
+
+
+def reindex_forcings(mask_map: PathLike, forcing: ForcingData, output_dir: PathLike = None) -> PathLike:
+    """Reindex forcing files to match mask map grid
+
+    Args:
+        mask_map: Path to NetCDF file used a boolean map that defines model boundaries.
+        forcing: Forcing data from ESMValTool
+        output_dir: Directory where to write the re-indexed files, given by user or created for user
+
+    Returns:
+        Output dir with re-indexed files.
+    """
+    output_dir = _generate_workdir(output_dir)
```

Comment:
I don't understand. Here you pass `output_dir` as the value for the `work_dir` argument. But if I read `_generate_workdir` correctly, it will then just return output_dir, right? So it will not do anything.

Reply:
right, the function `_check_work_dir` is now removed.